fix(app):  Fix ODD pressed state CSS

mjhuff opened the pull request:
closes RAUT-520
<!--
Thanks for taking the time to open a pull request! Please make sure you've read the "Opening Pull Requests" section of our Contributing Guide:

https://github.com/Opentrons/opentrons/blob/edge/CONTRIBUTING.md#opening-pull-requests

To ensure your code is reviewed quickly and thoroughly, please fill out the sections below to the best of your ability!
-->

# Overview

<!--
Use this section to describe your pull-request at a high level. If the PR addresses any open issues, please tag the issues here.
-->

This PR harmonizes the pressed state CSS styling various ODD buttons. Dark gray bounding boxes no longer appear, and pressed state is added to several pages that were missing it. There are also a couple of CSS fixes, like the long press ProtocolDashboard menu also showing pressed state for buttons not on the modal, and touchscreen sleep buttons having bounding boxes that did not fit the shape of the button. Additionally, the inconsistent UI behavior of pressed state for existing routes is fixed.

Note that certain hover effects will no longer appear when emulating the ODD on a computer. Also, I removed some .focus-visible properties, as these could not be triggered previously on the physical ODD and were causing UI bugs (rendered boxes not fitting the shape of the buttons).

### What Causes Grey Bounding Boxes?

The behavior occurs when the bounding box does not perfectly fit the size of the text, and any psuedoclass container other than active changes the cursor property to anything but default. Ex: `&.hover: {cursor: pointer;}`

The major culprit is the Btn primitive, which is now wrapped in a media query to prevent misuse, however, there are other custom buttons on various pages. These have been changed as well. 

In order to implement bounding boxes on icons (or anything else), active properties with cursor: default seem to work.

# Test Plan

<!--
Use this section to describe the steps that you took to test your Pull Request.
If you did not perform any testing provide justification why.

OT-3 Developers: You should default to testing on actual physical hardware.
Once again, if you did not perform testing against hardware, justify why.

Note: It can be helpful to write a test plan before doing development

Example Test Plan (HTTP API Change)

- Verified that new optional argument `dance-party` causes the robot to flash its lights, move the pipettes,
then home.
- Verified that when you omit the `dance-party` option the robot homes normally
- Added protocol that uses `dance-party` argument to G-Code Testing Suite
- Ran protocol that did not use `dance-party` argument and everything was successful
- Added unit tests to validate that changes to pydantic model are correct

-->

Must be tested on a physical ODD.
1. Long pressing the navbar links and overflow button on RobotDashboard should not cause grey bounding boxes to appear.
2. Navigating to Settings > Touchscreen Brightness and clicking the +/- should not cause grey bounding boxes to appear.
3. Settings > Touchscreen Sleep. Long pressing these buttons should not cause a visible gray bounding box that doesn't fit the dimensions of the buttons.
4. All Protocols. These buttons now have pressed state. Pinned protocols should also have pressed state.
5. All Protocols > Protocol (details page) > Start Setup (setup page). These buttons now have pressed state.
6.  Instruments Page. Pressed state is now consistent with other pressed state (previously was pressed for too long).
7. Start setup and then click instrument calibration. The wizard exit and back buttons should have state when pressed.
8. Network settings page now has pressed state for connected (green) networks and gray for unconnected networks.

# Changelog

<!--
List out the changes to the code in this PR. Please try your best to categorize your changes and describe what has changed and why.

Example changelog:
- Fixed app crash when trying to calibrate an illegal pipette
- Added state to API to track pipette usage
- Updated API docs to mention only two pipettes are supported

IMPORTANT: MAKE SURE ANY BREAKING CHANGES ARE PROPERLY COMMUNICATED
-->

- Fixed grey bounding boxes on existing pressed state.
- Harmonized behavior for existing pressed state.
- Created new pressed state behavior for buttons without it.

# Risk assessment

<!--
Carefully go over your pull request and look at the other parts of the codebase it may affect. Look for the possibility, even if you think it's small, that your change may affect some other part of the system - for instance, changing return tip behavior in protocol may also change the behavior of labware calibration.

Identify the other parts of the system your codebase may affect, so that in addition to your own review and testing, other people who may not have the system internalized as much as you can focus their attention and testing there.
-->
low

mjhuff committed:
fix(app): fix grey bounding boxes

Solves grey background boxes from appearing over certain buttons/navlinks when pressed on the
physical ODD.

mjhuff committed:
fix(app): add pressed state to ProtocolDetails and ProtocolSetup routes

mjhuff committed:
fix(app): fix sleep modal gray bounding boxes on physical ODD

Some CSS properties cause gray boxes that do not fit the dimensions of the buttons when pressed.
This bug only occurs on the ODD. Removing these CSS properties does not visibly impact styling.

mjhuff commented on app/src/organisms/Navigation/index.tsx:
```diff
@@ -157,25 +157,30 @@ const TouchNavLink = styled(NavLink)`
+  @media ${RESPONSIVENESS.touchscreenMediaQuerySpecs} {
+    cursor: default;
+
+    &:hover {
```
The hover effect doesn't impact the ODD here - I decided to leave it in.

mjhuff commented on app/src/organisms/Navigation/index.tsx:
```diff
@@ -157,25 +157,30 @@ const TouchNavLink = styled(NavLink)`
+    &:hover {
+      cursor: pointer;
+    }
+  }
```
The hover property doesn't affect the ODD here. I left it in for DX.

mjhuff commented on app/src/organisms/Navigation/index.tsx:
```diff
@@ -157,25 +157,30 @@ const TouchNavLink = styled(NavLink)`
+    &:hover {
+      cursor: pointer;
+    }
+  }
```
See other comment.

mjhuff committed:
Remove hover from overflow icon

koji commented on app/src/organisms/RobotSettingsDashboard/TouchscreenBrightness.tsx:
```diff
@@ -28,16 +27,6 @@ import {
 import type { Dispatch } from '../../redux/types'
 import type { SetSettingOption } from '../../pages/OnDeviceDisplay/RobotSettingsDashboard'
 
-const BUTTON_STYLE = css`
```
Actually we need to keep pseudo class for this button.
https://www.figma.com/file/OIdG64Q5cgvEw82ish5Eee/Design-System-(Flex)?node-id=715%3A18873&mode=dev

mjhuff replied:
@koji Great catch! Thanks for linking this.

koji commented on app/src/pages/OnDeviceDisplay/ProtocolSetup/index.tsx:
```diff
@@ -113,6 +113,18 @@ export function ProtocolSetupStep({
 
+  const PUSHED_STATE_STYLING = css`
+    &:active {
+      background-color: ${!disabled
+        ? status === 'general'
+          ? COLORS.darkBlack40
+          : status === 'ready'
+          ? COLORS.green3Pressed
+          : COLORS.yellow3Pressed
+        : ''};
+    }
```
Personally I like `Optional chaining` but we avoid using nested `Optional chaining`.
Maybe creating a new const and if-else/switch and passing the color value?

mjhuff committed:
fix(app): add more missing pressed state

Adds pressed state to Network settings buttons, pinned protocol cards, and select wizard buttons.

mjhuff committed:
Network settings 'connected' CSS tweak

mjhuff commented:
I've double checked all missing button pressed state, and we should be good to go! See the discussion on RAUT-520 for more of the specific missing pressed states that are resolved with this PR. Directly adjusting the Btn primitive solves buttons that implement it.

koji commented on app/src/organisms/ChildNavigation/index.tsx:
```diff
@@ -66,3 +68,20 @@ export function ChildNavigation({
+  &:disabled {
+    background-color: transparent;
+  }
+  @media ${RESPONSIVENESS.touchscreenMediaQuerySpecs} {
+    cursor: default;
+  }
```
question
Is this media-query for `dev-odd` mode?

mjhuff replied:
It impacts dev-odd if that's what you're asking. I noticed that on some pressed state, it was impacting the physical ODD, which is why I added it 👍

koji commented on app/src/organisms/RobotSettingsDashboard/NetworkSettings/index.tsx:
```diff
@@ -97,13 +98,22 @@ function NetworkSettingButton({
   networkName,
   onClick,
 }: NetworkSettingButtonProps): JSX.Element {
+  const PUSHED_STATE_STYLING = css`
```
Maybe STYLE since other styles that are defined by `css` use `_STYLE`.

```suggestion
  const PUSHED_STATE_STYLE = css`
```

koji reviewed (approved):
I left a couple of comments but the PR lgtm

mjhuff commented:
> I left a couple of comments but the PR lgtm

Thanks for checking all of this. I know it was a lot. Much appreciated!

mjhuff committed:
Updated CSS constants to be consistent with established patterns
